### DISCUSSION TITLE
Testing magma virtually

### DISCUSSION THREAD

**Discussion by fic23** [magma/magma#15153] [Q&A]
Hello everyone, I am a newbie in using the magma and also a newbie in the network community. I am trying to run and test the flow of the magma virtually, including the UEs, AGW and all components of the magma. I followed entirely the Quick Start guide [https://magma.github.io/magma/docs/basics/quick_start_guide](url) up until the NMS UI. What can I do to proceed with the testing phase? Can I run magma from A to Z virtually (that includes testing the whole traffic from UE to core network) without using any hardware? 
![image](https://user-images.githubusercontent.com/126511594/226129271-dd81918d-cb6b-4897-9498-a53aa5bb5c09.png)

**Comment by jordanvrtanoski** [comment]
You can run Magma completely virtually by running Orchestrator and AGW in a virtual machine and using a third virtual machine to simulate the RAN. You can use the SRS RAN simulator for 4G (LTE). I have prepared a repository to help you speed up the test. You can find the simulator on https://github.com/jordanvrtanoski/srsRAN-demo
